nautobot/extras/models/customfields.py

```diff
@@ -325,8 +328,97 @@ def validate(self, value):
 
             # Validate selected choice
             if self.type == CustomFieldTypeChoices.TYPE_SELECT:
-                if value not in self.choices:
-                    raise ValidationError(f"Invalid choice ({value}). Available choices are: {', '.join(self.choices)}")
+                if value not in self.choices.values_list("value", flat=True):
+                    raise ValidationError(
+                        f"Invalid choice ({value}). Available choices are: {', '.join(self.choices.values_list('value', flat=True))}"
+                    )
+
+            if self.type == CustomFieldTypeChoices.TYPE_MULTISELECT:
+                if not set(value).issubset(self.choices.values_list("value", flat=True)):
+                    raise ValidationError(
+                        f"Invalid choice(s) ({value}). Available choices are: {', '.join(self.choices.values_list('value', flat=True))}"
+                    )
 
         elif self.required:
             raise ValidationError("Required field cannot be empty.")
+
+    def delete(self, *args, **kwargs):
+        """
+        Handle the cleanup of old custom field data when a CustomField is deleted.
+        """
+        content_types = set(self.content_types.values_list("pk", flat=True))
+
+        super().delete(*args, **kwargs)
+
+        delete_custom_field_data.delay(self.name, content_types)
+
+
+class CustomFieldChoice(BaseModel):
+    """
+    The custom field choice is used to store the possible set of values for a selection type custom field
+    """
+
+    field = models.ForeignKey(
+        to="extras.CustomField",
+        on_delete=models.CASCADE,
+        related_name="choices",
+        limit_choices_to={"type": CustomFieldTypeChoices.TYPE_SELECT},
+    )
+    value = models.CharField(max_length=100)
+    weight = models.PositiveSmallIntegerField(default=100, help_text="Higher weights appear later in the list")
+
+    class Meta:
+        ordering = ["field", "weight", "value"]
+        unique_together = ["field", "value"]
+
+    def __str__(self):
+        return self.value
+
+    def clean(self):
+        if self.field.type not in (CustomFieldTypeChoices.TYPE_SELECT, CustomFieldTypeChoices.TYPE_MULTISELECT):
+            raise ValidationError("Custom field choices can only be assigned to selection fields.")
+
+    def save(self, *args, **kwargs):
+        """
+        When a custom field choice is saved, perform logic that will update data across all custom field data.
+        """
+        if self.present_in_database:
+            database_object = self.__class__.objects.get(pk=self.pk)
+        else:
+            database_object = self
+
+        super().save(*args, **kwargs)
+
+        if self.value != database_object.value:
+            update_custom_field_choice_data.delay(self.field.pk, database_object.value, self.value)
```

Comment:
This can throw a `rq.connections.NoRedisConnectionException` if you have an outdated `nautobot_config.py` that's defining `RQ_QUEUES` without `"custom_fields"` queue, as I've discovered. :-) Maybe add some error checks here?

Reply:
Yeah, good catch!

Reply:
I have addressed this with a new check E006